lib/hammer_cli_foreman/api/connection.rb

```diff
@@ -77,22 +135,53 @@ def build_default_params(settings, logger, locale)
         config[:timeout] = settings.get(:foreman, :request_timeout)
         config[:timeout] = -1 if (config[:timeout] && config[:timeout].to_i < 0)
         config[:apidoc_authenticated] = false
-        config[:authenticator] = create_authenticator(config[:uri], settings)
+        config[:authenticator] = create_authenticator(config[:uri], settings, auth_type)
         config
       end
     end
   end
 
-  CONNECTION_NAME = 'foreman'
-
   def self.foreman_api_connection
     HammerCLI.context[:api_connection].create(CONNECTION_NAME) do
       HammerCLIForeman::Api::Connection.new(HammerCLI::Settings, Logging.logger['API'], HammerCLI::I18n.locale)
     end
   end
 
+  def self.foreman_api_reconnect(auth_type)
+    HammerCLI.context[:api_connection].drop(CONNECTION_NAME)
+    if (session_file && !session_file.empty?)
+      File.delete(session_repo+"/"+session_file)
+    end
+    HammerCLI.context[:api_connection].create(CONNECTION_NAME) do
+      HammerCLIForeman::Api::Connection.new(
+        HammerCLI::Settings,
+        Logging.logger['API'],
+        HammerCLI::I18n.locale,
+        auth_type)
+    end
+  end
+
+
+  def self.session_contents
+    return {} unless (session_file && !session_file.empty?)
+    { :auth_type => JSON.parse(File.read(session_repo+"/"+session_file))["auth_type"],
+      :session_id =>JSON.parse(File.read(session_repo+"/"+session_file))["session_id"]}
+  end
+
+  def self.session_repo
+    session_path = Dir.home + "/.hammer/sessions"
+    Dir[session_path].first
+  end
+
+  def self.session_file
+    if Dir[session_repo+ "/*"].empty?
+      nil
+    else
+      Dir.entries(session_repo)[-1]
```

Comment:
This is not safe for a couple of reasons. I have multiple session files for multiple foremans I manage in my session dir. We should pick the correct one by url, otherwise we can end up reading wrong file. 
Also while this is not predictably sorted list I ended up with '..' on the last position resulting in ugly errors when reading the session. 

I wonder if it is time to represent the session file and operations with some class and have the usage unified across the codebase.
Would something with similar interface make sense?:
```ruby
session = HammerCLIForeman::Sessions.load(url) # load or create new session
expired = session.expired?
session.expire!
session.username
session.username = username
session.auth_type
session.auth_type = auth_type
session.save
session.id
session.id =  id
```

Reply:
Yeah! completely makes sense. I was actually assuming that a user will have only one session file and one foreman instance to manage. Thanks for bringing this up :)

Your suggestion about making a new class that takes care of the session file and operation completely makes sense.